preliminary network event work

Berthalamew opened the pull request:
- changes needed before pushing the network event implementation

Berthalamew committed:
misc changes

- separate private and public functions in cseries_windows_debug_pc

Berthalamew committed:
data changes

- rename s_data_array -> data_array and rename members
- add c_player_in_game_iterator and c_player_with_unit_iterator
- rename get_game_time_ticks -> game_time_get

Berthalamew committed:
rename network_address structure to transport_address

Berthalamew committed:
adjust cseries code

- add parameter IS_EXCEPTION in ASSERT_TRIGGER_EXCEPTION
- add csstrnlwr
- change vsprintf to use char* for args pointer

Berthalamew committed:
rename 	_game_team_none to _game_team_observer

Berthalamew committed:
move structures and enums out of players.h and into other files

Berthalamew committed:
header changes

- remove some header includes from files that weren't needed and take advantage of forward declarations in other places

Berthalamew committed:
add console_warning function

Berthalamew committed:
add c_network_link class

Berthalamew committed:
update c_network_time_statistics

- fix size
- add initialization function

Berthalamew committed:
add c_network_text_chat_manager

Berthalamew committed:
add various initialization functions for network code

Berthalamew committed:
add network configuration struct

- adjust network adapter code

Berthalamew committed:
remove players.h dependence in network_session.h

Kantanomo commented on xlive/Blam/Engine/interface/hud.cpp:
```diff
@@ -162,14 +162,15 @@ void hud_render_player_indicators(datum player_index)
 
-			bool is_enemy = game_engine_team_is_enemy((e_game_team)current_player->properties[0].team_index, (e_game_team)player->properties[0].team_index);
-			datum current_index = player_iterator.get_current_datum_index();
+			bool is_enemy = player->properties[0].team_index != player->properties[0].team_index;
```
Revert back to the game engine function, that is a class member function for game engines

Berthalamew replied:
changed the code to how it originally was

Berthalamew committed:
restore old hud indicator logic

- added back incase anyone wants to make use of it in multiplayer in the future

Berthalamew committed:
rename free to free_block in allocation functions
